tests/unit/query_v2_expression_evaluator.cpp

```diff
@@ -0,0 +1,1786 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <chrono>
+#include <cmath>
+#include <iterator>
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "coordinator/shard_map.hpp"
+#include "exceptions.hpp"
+#include "functions/awesome_memgraph_functions.hpp"
+#include "parser/opencypher/parser.hpp"
+#include "query/v2/accessors.hpp"
+#include "query/v2/bindings/eval.hpp"
+#include "query/v2/bindings/frame.hpp"
+#include "query/v2/context.hpp"
+#include "query/v2/frontend/ast/ast.hpp"
+#include "query/v2/requests.hpp"
+#include "query/v2/shard_request_manager.hpp"
+#include "storage/v3/property_value.hpp"
+#include "storage/v3/storage.hpp"
+#include "utils/exceptions.hpp"
+#include "utils/string.hpp"
+
+#include "query_v2_query_common.hpp"
+//#include "utils/temporal.hpp"
+
+using namespace memgraph::query::v2;
+using memgraph::query::v2::test_common::ToIntList;
+using testing::ElementsAre;
+using testing::UnorderedElementsAre;
+
+using memgraph::io::Time;
+using memgraph::io::TimedOut;
+using memgraph::io::simulator::Simulator;
+using memgraph::io::simulator::SimulatorConfig;
+using memgraph::io::simulator::SimulatorTransport;
+using memgraph::msgs::CreateExpandRequest;
+using memgraph::msgs::CreateExpandResponse;
+using memgraph::msgs::CreateVerticesRequest;
+using memgraph::msgs::CreateVerticesResponse;
+using memgraph::msgs::ExpandOneRequest;
+using memgraph::msgs::ExpandOneResponse;
+using memgraph::msgs::ExpandOneResultRow;
+using memgraph::msgs::NewExpand;
+using memgraph::msgs::NewVertex;
+using memgraph::msgs::ScanVerticesRequest;
+using memgraph::msgs::ScanVerticesResponse;
+using ShardMap = memgraph::coordinator::ShardMap;
+using LabelId = memgraph::storage::v3::LabelId;
+using PropertyId = memgraph::storage::v3::PropertyId;
+using memgraph::coordinator::Shards;
+using CompoundKey = memgraph::coordinator::PrimaryKey;
+using memgraph::expr::ExpressionRuntimeException;
+using memgraph::functions::FunctionRuntimeException;
+
+namespace {
```

Comment:
And you can drop all memgraph::query::v2 namespacing below
```suggestion
namespace memgraph::query::v2::tests {
```